src/fields/fft_poisson_solver/fft/WrapRocDST.cpp

```diff
@@ -0,0 +1,190 @@
+                 const int n_data, const int n_batch)
+    {
+        HIPACE_PROFILE("AnyDST::ToSine()");
+        /* todo */
+    }
+
+    /** \brief Transpose input matrix
+     * out[idy][idx] = in[idx][idy]
+     *
+     * \param[in] in input real array
+     * \param[out] out output real array
+     * \param[in] n_data number of (contiguous) rows in input matrix
+     * \param[in] n_batch number of (strided) columns in input matrix
+     */
+    void Transpose (const amrex::Real* const AMREX_RESTRICT in,
+                    amrex::Real* const AMREX_RESTRICT out,
+                    const int n_data, const int n_batch)
+    {
+        HIPACE_PROFILE("AnyDST::Transpose()");
+        /* todo */
+    }
+
+    DSTplan CreatePlan (const amrex::IntVect& real_size, amrex::FArrayBox* position_array,
+                        amrex::FArrayBox* fourier_array)
+    {
+        HIPACE_PROFILE("AnyDST::CreatePlan()");
+        DSTplan dst_plan;
+        const int nx = real_size[0];
+        const int ny = real_size[1];
+        rocfft_status status;
+        int dim = 2;
+
+        // Allocate expanded_position_array Real of size (2*nx+2, 2*ny+2)
+        // Allocate expanded_fourier_array Complex of size (nx+2, 2*ny+2)
+        amrex::Box expanded_position_box {{0, 0, 0}, {2*nx+1, 2*ny+1, 0}};
+        amrex::Box expanded_fourier_box {{0, 0, 0}, {nx+1, 2*ny+1, 0}};
+        dst_plan.m_expanded_position_array =
+            std::make_unique<amrex::FArrayBox>(
+                expanded_position_box, 1);
+        dst_plan.m_expanded_fourier_array =
+            std::make_unique<amrex::BaseFab<amrex::GpuComplex<amrex::Real>>>(
+                expanded_fourier_box, 1);
+
+        // setting the initial values to 0
+        // we don't set the expanded Fourier array, because it will be initialized by the FFT
+        dst_plan.m_expanded_position_array->setVal<amrex::RunOn::Device>(0.,
+            dst_plan.m_expanded_position_array->box(), 0,
+            dst_plan.m_expanded_position_array->nComp());
+
+        // check for type of expanded size, should be const size_t *
+        const amrex::IntVect& expanded_size = expanded_position_box.length();
+        const std::size_t lengths[] = {AMREX_D_DECL(std::size_t(expanded_size[0]),
+                                                    std::size_t(expanded_size[1]),
+                                                    std::size_t(expanded_size[2]))};
+
+        // Initialize fft_plan.m_plan with the vendor fft plan.
+        rocfft_status result;
+        result = rocfft_plan_create(&(dst_plan.m_plan), \
+                                    rocfft_placement_notinplace, \
+                                    rocfft_transform_type_real_forward, \
+                                    precision, \
+                                    dim, \
+                                    lengths, \
+                                    1, \
+                                    nullptr);
```

Comment:
Style note: line continuations with `\` are not needed here.
```suggestion
        result = rocfft_plan_create(&(dst_plan.m_plan),
                                    rocfft_placement_notinplace,
                                    rocfft_transform_type_real_forward,
                                    precision,
                                    dim,
                                    lengths,
                                    1,
                                    nullptr);
```